### PR TITLE
Capacity refactor

### PR DESCRIPTION
**Pull request by 0xmountaintop**
## 1. Purpose or design rationale of this PR

...

## 2. PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [ ] build: Changes that affect the build system or external dependencies (example scopes: yarn, eslint, typescript)
- [ ] ci: Changes to our CI configuration files and scripts (example scopes: vercel, github, cypress)
- [ ] docs: Documentation-only changes
- [ ] feat: A new feature
- [ ] fix: A bug fix
- [ ] perf: A code change that improves performance
- [ ] refactor: A code change that doesn't fix a bug, or add a feature, or improves performance
- [ ] style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- [ ] test: Adding missing tests or correcting existing tests

## 3. Deployment tag versioning

Has the version in `params/version.go` been updated?

- [ ] This PR doesn't involve a new deployment, git tag, docker image tag, and it doesn't affect traces
- [ ] Yes

## 4. Breaking change label

Does this PR have the `breaking-change` label?

- [ ] This PR is not a breaking change
- [ ] Yes

### CODE REVIEW

**Review by silathdiir — APPROVED** [review]
LGTM 👍

File: rollup/circuitcapacitychecker/impl.go [hunk 1]
```diff
@@ -17,15 +17,26 @@ import (
 	"github.com/scroll-tech/go-ethereum/log"
 )
 
-type CircuitCapacityChecker struct{}
+// global count for total CircuitCapacityChecker
+var cnt uint8
```
Comment (silathdiir) [inline rollup/circuitcapacitychecker/impl.go]: Sorry, just find out if we could do this fix.
Should we add a new function as `circuit_capacity_checker_count` (or other function name `last_circuit_capacity_checker_id`?) to return `CHECKERS.len()` from Rust and delete this global count in go side?

**Review by Lawliet-Chan — COMMENTED** [review]
what is the `circuit_capacity`? could you describe it?

**Review by Lawliet-Chan — COMMENTED** [review]
This PR should add  files `Cargo.toml` and `rust-toolchain`

**Review by Lawliet-Chan — APPROVED** [review]

File: rollup/circuitcapacitychecker/impl.go (hunk 1 above)
Comment (0xmountaintop) [reply rollup/circuitcapacitychecker/impl.go]: good idea. I will fix it

Comment (0xmountaintop) [reply rollup/circuitcapacitychecker/impl.go]: fixed in https://github.com/scroll-tech/go-ethereum/pull/374/commits/1284caf864d0cba52b7ae60234c97df0006cfaa5

**Review by silathdiir — APPROVED** [review]

**Pull request merged**